stock_request/models/stock_request_order.py

```diff
@@ -144,6 +146,20 @@ def _get_default_requested_by(self):
         ("name_uniq", "unique(name, company_id)", "Stock Request name must be unique")
     ]
 
+    @api.depends("stock_request_ids.state")
+    def _compute_state(self):
+        for item in self:
+            if not item.stock_request_ids or all(
+                x.state == "draft" for x in item.stock_request_ids
+            ):
+                item.state = "draft"
+            elif all(x.state == "cancel" for x in item.stock_request_ids):
+                item.state = "cancel"
+            elif all(x.state in ("done", "cancel") for x in item.stock_request_ids):
+                item.state = "done"
```

Comment:
"Cache" the states for not having to query them several times:

```suggestion
            states = item.stock_request_ids.mapped("state")
            if not item.stock_request_ids or all(x == "draft" for x in states):
                item.state = "draft"
            elif all(x == "cancel" for x in states):
                item.state = "cancel"
            elif all(x in ("done", "cancel") for x in states):
                item.state = "done"
```